gef.py

```diff
@@ -2699,9 +2698,13 @@ def to_unsigned_long(v):
     return int(v.cast(gdb.Value(mask).type)) & mask
 
 
-@lru_cache()
 def get_register(regname):
     """Return a register's value."""
+    return get_register_for_selected_frame(id(gdb.selected_frame()), regname)
```

Comment:
Have you confirmed that `gdb.selected_frame()`, if called more than once, returns the same object?

Reply:
Thank you for review. Unfortunately, no. It changes.
Maybe just remove `lru_cache` for `get_register` in such case?

Reply:
That would slow things down a lot. Maybe we can use `gdb.selected_frame().pc()`? Although that could break on recursive functions. We should play with the frame object to see if there something we can use.